Sources/EdgeNetworkHandlers/EdgeNetworkService.swift

```diff
@@ -110,34 +114,39 @@ class EdgeNetworkService {
                                        streaming: requestBody.meta?.konductorConfig?.streaming,
                                        responseCallback: responseCallback)
                     responseCallback.onComplete()
-                    completion(true) // successful request, return true
+                    completion(true, 0.0) // successful request, return true
                 } else if responseCode == HttpResponseCodes.noContent.rawValue {
                     // Successful collect requests do not return content
                     Log.debug(label: self.LOG_TAG, "doRequest - Collect connection to Experience Edge was successful.")
                     responseCallback.onComplete()
-                    completion(true) // successful request, return true
+                    completion(true, 0.0) // successful request, return true
                 } else if self.recoverableNetworkErrorCodes.contains(responseCode) {
                     Log.debug(label: self.LOG_TAG, "doRequest - Connection to Experience Edge returned recoverable error code \(responseCode)")
-                    completion(false) // failed, but recoverable so retry
+                    let retryHeader = connection.responseHttpHeader(forKey: EdgeConstants.NetworkKeys.HEADER_KEY_RETRY_AFTER)
+                    var retryInterval = EdgeConstants.NetworkKeys.RETRY_INTERVAL
+                    if let retryHeader = retryHeader, let retryAfterInterval = TimeInterval(retryHeader) {
```

Comment:
```
var retryAfterInterval = nil
if let retryHeader = connection.responseHttpHeader(forKey: EdgeConstants.NetworkKeys.HEADER_KEY_RETRY_AFTER) 
    retryAfterInterval = TimeInterval(retryHeader)
}
completion(false, retryInterval)

Reply:
I'm exactly sure what the suggestion is. But the current code snippet above doesn't compile.  Mainly due to the fact that `TimeInterval` is a typealias of a `Double`, so it can't take the value of nil and the completion `(Bool, TimerInterval)` can't support passing in `(boolValue, nil)`, so we must provide a default value.

Reply:
As stated above `TimeInterval` cannot be `nil`. It is a type alias of `Double` which cannot take the value of `nil`.